consumer.go

```diff
@@ -17,21 +22,21 @@ type Consumer interface {
 	// which this consumer belongs.
 	SetGroupID(groupID string)
 
-	// RunTopic runs a message handler against a topic and partition. The handler
+	// RunTopic runs a message handler against a topic. The handler
 	// gets called each time a new message is received.
-	RunTopic(topic string, partition int, handler Handler)
+	RunTopic(topic string, handler Handler)
 }
 
 type consumer struct {
 	config *go_kafka_client.ConsumerConfig
 }
 
-func NewConsumer(zookeeper string, schemaRepo string) (Consumer, error) {
+func NewConsumer(zookeeper string, schemaRepo string, offsetStrategy string) (Consumer, error) {
 	zConfig := go_kafka_client.NewZookeeperConfig()
 	zConfig.ZookeeperConnect = []string{zookeeper}
 
 	consumerConfig := go_kafka_client.DefaultConsumerConfig()
-	consumerConfig.AutoOffsetReset = go_kafka_client.LargestOffset
+	consumerConfig.AutoOffsetReset = offsetStrategy
```

Comment:
Can you add some validation to make sure that we're enforcing that `offsetStrategy` is either `OffsetResetSmallest` or `OffsetResetLargest`?